test/suites/consolidation/suite_test.go

```diff
@@ -0,0 +1,866 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package consolidation_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/awslabs/operatorpkg/object"
+	"github.com/samber/lo"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	coretest "sigs.k8s.io/karpenter/pkg/test"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1alpha2"
+	"github.com/Azure/karpenter-provider-azure/test/pkg/debug"
+	"github.com/Azure/karpenter-provider-azure/test/pkg/environment/azure"
+	"github.com/Azure/karpenter-provider-azure/test/pkg/environment/common"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var env *azure.Environment
+
+func TestConsolidation(t *testing.T) {
+	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		env = azure.NewEnvironment(t)
+	})
+	AfterSuite(func() {
+		env.Stop()
+	})
+	RunSpecs(t, "Consolidation")
+}
+
+var nodeClass *v1alpha2.AKSNodeClass
+
+var _ = BeforeEach(func() {
+	nodeClass = env.DefaultAKSNodeClass()
+	env.BeforeEach()
+})
+var _ = AfterEach(func() { env.Cleanup() })
+var _ = AfterEach(func() { env.AfterEach() })
+
+var _ = Describe("Consolidation", Ordered, func() {
+	Context("LastPodEventTime", func() {
+		var nodePool *karpv1.NodePool
+		BeforeEach(func() {
+			nodePool = env.DefaultNodePool(nodeClass)
+			nodePool.Spec.Disruption.ConsolidateAfter = karpv1.MustParseNillableDuration("Never")
+		})
+		It("should update lastPodEventTime when pods are scheduled and removed", func() {
```

Comment:
Should we be testing lastPodEventTime repeatedly scheduling pods on the same node to ensure its being updated consistently for the same nodes?

Reply:
Yes - and that's exactly what this test does. What am I missing?